File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/DailyCaloriesMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.model.DailyCalories
import com.rookmotion.rook_sdk_health_connect.proto.DailyCaloriesProto

fun DailyCalories.toProto(): DailyCaloriesProto {
    return DailyCaloriesProto.newBuilder()
        .setBasal(basal)
        .setActive(active)
        .build()
}

File: packages/rook_sdk_health_connect/lib/src/data/mapper/daily_calories_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_health_connect/src/data/proto/protos.pb.dart';

extension DailyCaloriesMappers on DailyCaloriesProto {
  DailyCalories toDomain() {
    return DailyCalories(basal: basal, active: active);
  }
}
